### PR TITLE
User autocomplete

### PR DESCRIPTION
**Pull request by williamnf**
Fixes #

### Proposed fixes:
Gives suggestion of users while adding contributors.
Changed contributors interface.

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [X] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

File: src/views/layout/includes.ejs [hunk 1]
```diff
@@ -117,10 +117,10 @@
     <script type="text/javascript" src="/app/js/controllers/social/file_versions_controller.js"></script><!--NELSON-->
     <script type="text/javascript" src="/app/js/controllers/notifications/notification_controller.js"></script><!--NELSON-->
 
+    <script type="text/javascript" src="/app/js/controllers/project/admin/project_contributor_controller.js"></script>
```
Comment (silvae86) [inline src/views/layout/includes.ejs]: Acho que se devia fundir estes dois controladores num único project_admin_controller.js que seria usado na div que contém **todas** as tabs. Assim cobriria tanto uma tab como a outra num só scope.

File: src/views/projects/administration/administer.ejs [hunk 2]
```diff
@@ -98,62 +103,43 @@
                 </div>
             </div>
 
-            <div class="row">
-                <div class="col-lg-12">
+            <div ng-controller="UserListCtrl" class="row">
+                <div ng-init="get_contributors('<%=locals.project.dcterms.contributor%>')" class="col-lg-12">
```
Comment (silvae86) [inline src/views/projects/administration/administer.ejs]: Seria melhor chamar uma função init() e essa sim chamaria o get_contributors. Há mesmo necessidade de passar este argumento? Talvez fosse melhor criar um ponto na API para obter os contribuidores atuais e evitar isto.

File: src/views/projects/administration/administer.ejs [hunk 3]
```diff
@@ -98,62 +103,43 @@
...
-                            }
-                            %>
-                            <tr>
+
+                            <tr ng-repeat="contributor in contributors track by $index">
                                 <td>
                                     <div class="row">
                                         <div class="col-xs-10">
-                                            <input type="text" class="form-control input-large" placeholder="New Contributor URI" name="newContributor">
+                                            <%- include users_autocomplete.ejs%>
```
Comment (silvae86) [inline src/views/projects/administration/administer.ejs]: Gostei da reutilização do autocomplete e separação num ficheiro distinto.

**Comment by williamnf** [comment]
@NelsonPereira1991 ja fiz o pr se quiseres ver no [projects.js](https://github.com/feup-infolab/dendro/blob/userAutocomplete/src/controllers/projects.js)

**Comment by silvae86** [comment]
To Fix:

1. Removing a user does not update the list. 
2. When you try to add a user that does not exist, the message is invalid

The interface is not clear to the user, since it is not obvious that the things are not saved until the user clicks the Update button. Suggestions:

1. Put the "Add new Contributor" at the top instead of bottom. 
2. Update Button -> Change to "Save changes" and add a Save icon font awesome (http://fontawesome.io/icon/floppy-o/)

screen shot 2017-03-20 at 10 39 26
screen shot 2017-03-06 at 18 06 06

**Pull request merged**